### PR TITLE
Comments: update data-layer to use new comments-tree endpoint

### PR DESCRIPTION
**Pull request by gwwar**
This PR updates the data-layer to point at `https://public-api.wordpress.com/rest/v1/sites/{$blog_id}/comments-tree` instead of  `https://public-api.wordpress.com/wpcom/v2/sites/{$blog_id}/comments-tree`. I've also updated the folder location to match convention and added missing tests.

We should follow up on this PR to update the `CommentList` component so Jetpack Sites >= 5.3 can also use this endpoint.

### Testing Instructions
- Navigate to /comments
- Comments behave normally for Simple, Jetpack and Atomic Sites

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/comments-tree/index.js [hunk 1]
```diff
@@ -17,15 +17,15 @@ import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
...
+export const fetchCommentsTreeForSite = ( { dispatch }, action ) => {
 	const { siteId, status = 'unapproved' } = action.query;
 
 	dispatch(
 		http(
 			{
 				method: 'GET',
 				path: `/sites/${ siteId }/comments-tree`,
-				apiNamespace: 'wpcom/v2',
+				apiVersion: '1',
```
Comment (dmsnell) [inline client/state/data-layer/wpcom/sites/comments-tree/index.js]: should this not be `v1`?

Comment (gwwar) [inline client/state/data-layer/wpcom/sites/comments-tree/index.js]: As an api namespace? or does apiVersion also accept a leading `v`?

For reference the old endpoint is at wp rest api with the wpcom/v2 namespace:
screen shot 2017-08-19 at 12 32 58 pm

vs wp.com api at version 1

screen shot 2017-08-19 at 12 32 01 pm

Comment (dmsnell) [inline client/state/data-layer/wpcom/sites/comments-tree/index.js]: just that I thought you had to manually at the `v` when using `apiVersion`…

Comment (gwwar) [inline client/state/data-layer/wpcom/sites/comments-tree/index.js]: I didn't find any code usages doing that. Note that the default version is also set without a leading `v` https://github.com/Automattic/wpcom.js/blob/master/index.js#L81

The request also looks correct.
screen shot 2017-08-19 at 4 40 01 pm

**Comment by gwwar** [comment]
Thanks for the reviews @vindl @dmsnell !

**Pull request merged**